### PR TITLE
Fix bugs when version is None and simplify naming of variables

### PR DESCRIPTION
**Pull request by tuxdna**
For: https://github.com/openshiftio/openshift.io/issues/2219

### CODE REVIEW

File: src/data_importer.py [hunk 1]
```diff
@@ -68,11 +68,11 @@ def _import_keys_from_s3_http(data_source, epv_list):
...
                 report['message'] = 'Nothing to be imported! No data found on S3 to be imported!'
                 continue
-            obj = {
-                'ecosystem': contents.get('ecosystem'),
-                'package': contents.get('package'),
-                'version': contents.get('version')
-            }
+            pkg_ecosystem = contents.get('ecosystem')
+            pkg_name = contents.get('package')
+            pkg_version = contents.get('version', '') or ''
```
Comment (jpopelka) [inline src/data_importer.py]: a bit simplified `contents.get('version') or ''`
should work the same

File: src/data_importer.py [hunk 2]
```diff
@@ -161,49 +160,58 @@ def _log_report_msg(import_type, report):
...
-                pkg_list_keys.extend(data_source.list_files(bucket_name=config.AWS_PKG_BUCKET,
-                                                            prefix=pkg_key_prefix))
-
-            if 'version' in epv and epv.get('version') is not None:
+
+            pkg_data = {'package': epv_name, 'version': epv_version, 'ecosystem': epv_ecosystem}
+
+            # Get Package level keys
+            package_prefix = version_prefix = (epv.get('ecosystem') + "/" + epv.get('name') +
+                                               "/")
```
Comment (jpopelka) [inline src/data_importer.py]:
```python
package_prefix = version_prefix = epv_ecosystem + "/" + epv_name + "/"
```

File: src/data_importer.py [hunk 3]
```diff
@@ -161,49 +160,58 @@ def _log_report_msg(import_type, report):
...
+
+            pkg_list_keys = data_source.list_files(bucket_name=config.AWS_PKG_BUCKET,
+                                                   prefix=package_prefix)
+
+            ver_list_keys = []
+            if epv_version:
                 # Get EPV level keys
-                ver_key_prefix = (epv.get('ecosystem') + "/" + epv.get('name') + "/" +
+                version_prefix = (epv.get('ecosystem') + "/" + epv.get('name') + "/" +
                                   epv.get('version'))
```
Comment (jpopelka) [inline src/data_importer.py]:
```python
version_prefix = epv_ecosystem + "/" + epv_name + "/" + epv_version
```

File: src/data_importer.py [hunk 4]
```diff
@@ -161,49 +160,58 @@ def _log_report_msg(import_type, report):
...
-                'pkg_list_keys': pkg_list_keys,
-                'pkg_key_prefix': pkg_key_prefix,
-                'package': epv.get('name'),
-                'version': epv.get('version', ''),
-                'ecosystem': epv.get('ecosystem')}}
+            # store s3 object paths for this epv
+            pkg_data['ver_key_prefix'] = version_prefix
+            pkg_data['ver_list_keys'] = ver_list_keys
+            pkg_data['pkg_key_prefix'] = package_prefix
+            pkg_data['pkg_list_keys'] = pkg_list_keys
```
Comment (jpopelka) [inline src/data_importer.py]: You define `pkg_data` quite early in the function, but use it for the first time here. What about simply:
```python
pkg_data = {
'package': epv_name,
'version': epv_version,
'ecosystem': epv_ecosystem,
'ver_key_prefix': version_prefix,
'ver_list_keys': ver_list_keys,
'pkg_key_prefix': package_prefix,
'pkg_list_keys': pkg_list_keys
}
```
here

File: src/data_importer.py (hunk 1 above)
Comment (tuxdna) [reply src/data_importer.py]: This applies for the case when `version` is not in dictionary, and also when `version` is in dictionary but with `None` value.

File: src/data_importer.py (hunk 2 above)
Comment (tuxdna) [reply src/data_importer.py]: Sure, will fix this.

File: src/data_importer.py (hunk 4 above)
Comment (tuxdna) [reply src/data_importer.py]: Sure, will fix.

File: src/data_importer.py (hunk 1 above)
Comment (jpopelka) [reply src/data_importer.py]: - If `version is not in contents`, the `contents.get('version')` returns `None` and `None or ''` is `''`
- If `contents['version'] == None`, the `contents.get('version')` also returns `None` and `None or ''` is `''`

Comment (tuxdna) [reply src/data_importer.py]: @jpopelka You are right. I missed that `get()` returns `None` in absence of given key.

**Comment by tuxdna** [comment]
Thanks @jpopelka :+1:

**Pull request merged**